casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -623,19 +623,20 @@ object BlockAPI {
   )(implicit casper: MultiParentCasper[F]): F[A] =
     for {
       dag <- casper.blockDag
-      // TODO this is temporary solution to not calculate fault tolerance for old blocks which is costly
-      oldBlock  = dag.latestBlockNumber.map(_ - block.body.state.blockNumber).map(_ > 100)
-      isInvalid = dag.lookup(block.blockHash).map(_.get.invalid)
+      // TODO this is temporary solution to not calculate fault tolerance all the blocks
+      oldBlock = dag.latestBlockNumber.map(_ - block.body.state.blockNumber).map(_ > 100)
       // Old block fault tolerance / invalid block has -1.0 fault tolerance
-      oldBlockFaultTolerance = isInvalid.map(if (_) -1f else 1f)
       normalizedFaultTolerance <- oldBlock.ifM(
-                                   oldBlockFaultTolerance,
+                                   dag
+                                     .isFinalized(block.blockHash)
+                                     .map(isFinalized => if (isFinalized) 1f else -1f),
```

Comment:
I was thinking to eliminate any calculation for finalized blocks and calculate only if not finalized and not old.

For example if `/blocks/100` is called, most blocks is finalized and we can save ~90 calculations.

Reply:
On dev standup we discussed that fault tolerance for finalized blocks can have value other then `1` so it make sense to calculate always latest blocks.